packages/block-library/src/column/edit.js

```diff
@@ -14,22 +14,43 @@ import {
 	__experimentalBlock as Block,
 } from '@wordpress/block-editor';
 import { PanelBody, RangeControl } from '@wordpress/components';
-import { withDispatch, withSelect } from '@wordpress/data';
-import { compose } from '@wordpress/compose';
+import { useSelect, useDispatch } from '@wordpress/data';
 import { __ } from '@wordpress/i18n';
 
 function ColumnEdit( {
-	attributes,
+	attributes: { verticalAlignment, width },
 	setAttributes,
-	updateAlignment,
-	hasChildBlocks,
+	clientId,
 } ) {
-	const { verticalAlignment, width } = attributes;
-
 	const classes = classnames( 'block-core-columns', {
 		[ `is-vertically-aligned-${ verticalAlignment }` ]: verticalAlignment,
 	} );
 
+	const { hasChildBlocks, rootClientId } = useSelect(
+		( select ) => {
+			const { getBlockOrder, getBlockRootClientId } = select(
+				'core/block-editor'
+			);
+
+			return {
+				hasChildBlocks: getBlockOrder( clientId ).length > 0,
+				rootClientId: getBlockRootClientId( clientId ),
+			};
+		},
+		[ clientId ]
+	);
+
+	const { updateBlockAttributes } = useDispatch( 'core/block-editor' );
+
+	const updateAlignment = ( value ) => {
+		// Update own alignment.
+		setAttributes( { verticalAlignment: value } );
+		// Reset Parent Columns Block
```

Comment:
While we're doing this refactor, we can tweak this comment to match our guidelines:

```suggestion
		// Reset parent Columns block.
```